thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/components/PercentageChangeRuleDetector.java

```diff
@@ -76,16 +80,34 @@
 
   @Override
   public DetectionResult runDetection(Interval window, String metricUrn) {
+    long startMillis = window.getStartMillis();
+
+    // align start day to the user specified week start
+    if (Objects.nonNull(this.weekStart)) {
+      startMillis = window.getStart().withDayOfWeek(weekStart.getValue()).minusWeeks(1).getMillis();
+    }
+
     MetricEntity me = MetricEntity.fromURN(metricUrn);
-    MetricSlice slice = MetricSlice.from(me.getId(), window.getStartMillis(), window.getEndMillis(), me.getFilters(), timeGranularity);
+    MetricSlice slice = MetricSlice.from(me.getId(), startMillis, window.getEndMillis(), me.getFilters(), timeGranularity);
     List<MetricSlice> slices = new ArrayList<>(this.baseline.scatter(slice));
     slices.add(slice);
 
     InputData data = this.dataFetcher.fetchData(new InputDataSpec().withTimeseriesSlices(slices)
         .withMetricIdsForDataset(Collections.singletonList(slice.getMetricId())));
-    DataFrame dfCurr = data.getTimeseries().get(slice).renameSeries(COL_VALUE, COL_CURR);
     DataFrame dfBase = this.baseline.gather(slice, data.getTimeseries());
+    DataFrame dfCurr = data.getTimeseries().get(slice);
+    DatasetConfigDTO datasetConfig = data.getDatasetForMetricId().get(me.getId());
 
+    // aggregate data to specified weekly granularity
+    if (this.monitoringGranularity.endsWith("WEEKS")) {
```

Comment:
Do we also support MONTHS?

Reply:
MONTHS is supported only if the metric itself is a monthly metric. The aggregation to monthly is not supported. I think that's out of the scope of this PR.

Reply:
Sure. Thanks.